sensing/autoware_pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp

```diff
@@ -258,6 +260,25 @@ void DistortionCorrector<T>::undistortPointCloud(
     // Undistort a single point based on the strategy
     undistortPoint(it_x, it_y, it_z, it_twist, it_imu, time_offset, is_twist_valid, is_imu_valid);
 
+    if (!sensor_azimuth_coordinate.empty() && pointcloudTransformNeeded()) {
+      // Input frame should be in the sensor frame.
+      *it_distance = sqrt(*it_x * *it_x + *it_y * *it_y + *it_z * *it_z);
+      float cartesian_coordinate_azimuth = cv::fastAtan2(*it_y, *it_x);
+      if (sensor_azimuth_coordinate == "velodyne") {
+        *it_azimuth = (360 - cartesian_coordinate_azimuth) * M_PI / 180;
+      } else if (sensor_azimuth_coordinate == "hesai") {
+        *it_azimuth = (90 - cartesian_coordinate_azimuth) < 0
+                        ? (90 - cartesian_coordinate_azimuth + 360) * M_PI / 180
+                        : (90 - cartesian_coordinate_azimuth) * M_PI / 180;
+      } else {
+        throw std::runtime_error(
+          sensor_azimuth_coordinate + " azimuth coordinate is not supported");
+      }
```

Comment:
Why is this necessary for each vendor? In Autoware, we prioritize designing modules to be adaptive, so we want to avoid vendor-specific processing in the code of this node as much as possible.

Reply:
Thanks for your comment, because the definition of azimuth depends on each vendor like the image below. 
![image](https://github.com/user-attachments/assets/36c8ed58-e8c0-4993-b1f1-20e6327b5f3e)

reference: https://github.com/vividf/autoware.universe/blob/feature/distortion_corrector_node_update_azimuth_and_distance/sensing/autoware_pointcloud_preprocessor/docs/distortion-corrector.md

Reply:
Do you have any suggestions for this? Thanks!

Reply:
@vividf I see 👍 Thank you for the quick response. I think it would be better to handle the differences in azimuth definitions within the Nebula decoder so that the preprocessor node doesn't have to account for these differences. What do you think?

Reply:
@yukkysaito Thanks for your advice! I talked about this with @drwnz in today's meeting.

Reply:
@yukkysaito 
In the end, we decided to implement a conversion formula to adapt to different azimuth coordinate systems for this node.
Additionally, there is a big chance that we will also unify the azimuth in Nebula

Reply:
@vividf Thank you for your response. Which part of the code are you referring to?

Reply:
@yukkysaito 
This function calculate the conversion formula https://github.com/autowarefoundation/autoware.universe/blob/a3c977b5b7772cb6e83e183e80ec9c9b2894f48e/sensing/autoware_pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp#L192